src/DataTransferObject.php

```diff
@@ -82,10 +82,40 @@ public function except(string ...$keys): DataTransferObject
     public function toArray(): array
     {
         if (count($this->onlyKeys)) {
-            return Arr::only($this->all(), $this->onlyKeys);
+            $array = Arr::only($this->all(), $this->onlyKeys);
+        } else {
+            $array = Arr::except($this->all(), $this->exceptKeys);
         }
 
-        return Arr::except($this->all(), $this->exceptKeys);
+        foreach ($array as $key => $property) {
+            $array[$key] = $this->convertForArray($property);
+        }
+
+        return $array;
+    }
+
+    /**
+     * @param mixed $property
+     *
+     * @return mixed
+     */
+    protected function convertForArray($property)
+    {
+        if ($property instanceof DataTransferObject) {
+            return $property->toArray();
+        }
+
+        if (is_object($property) && method_exists($property, 'toArray')) {
```

Comment:
Should we allow this much magic? My guess is that you have Laravel projects in mind, but I think making the assumption that everything should be cast to an array can be quite dangerous..

Reply:
I was think about this. In my personal cases it will be very useful. But for common purposes it maybe very unpredictable. It will be removed